extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -101,16 +142,52 @@ public void initialize() {
         }
 
         DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
-        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        builder.setPskStore(new StaticPskStore(configuration.identity, configuration.preSharedKey.getBytes()));
         dtlsConnector = new DTLSConnector(builder.build());
         endPoint = new TradfriCoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
         deviceClient.setEndpoint(endPoint);
         updateStatus(ThingStatus.UNKNOWN);
 
         // schedule a new scan every minute
-        scanJob = scheduler.scheduleWithFixedDelay(() -> {
-            startScan();
-        }, 0, 1, TimeUnit.MINUTES);
+        scanJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, 1, TimeUnit.MINUTES);
+    }
+
+    protected String authenticateOnGateway(TradfriGatewayConfig configuration) {
+        String preSharedPsk = null;
+        CoapResponse gatewayResponse;
+        try {
+            DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
+            builder.setPskStore(new StaticPskStore("Client_identity", configuration.code.getBytes()));
+
+            DTLSConnector dtlsConnector = new DTLSConnector(builder.build());
+            CoapEndpoint authEndpoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+            String authUrl = "coaps://" + configuration.host + ":" + configuration.port + "/15011/9063";
+
+            CoapClient deviceClient = new CoapClient(new URI(authUrl));
+            deviceClient.setEndpoint(authEndpoint);
+
+            JsonObject json = new JsonObject();
+            json.addProperty(CLIENT_IDENTITY_PROPOSED, configuration.identity);
+
+            gatewayResponse = deviceClient.post(json.toString(), 0);
+
+            authEndpoint.destroy();
+            deviceClient.shutdown();
+
+            if (gatewayResponse.isSuccess()) {
+                String responseText = gatewayResponse.getResponseText();
+                json = new JsonParser().parse(responseText).getAsJsonObject();
+                preSharedPsk = json.get(NEW_PSK_BY_GW).getAsString();
+            } else {
+                logger.error("Can't obtain pre-shared key for identity => {} (response code => {}, response text => {})",
```

Comment:
This does not qualify for an error logging as it is an expected path from an ESH perspective and not to be considered a bug on our end.
Please also note that `xxx => {}` is not a common pattern in our code. We usually use `xxx '{}'`.

Reply:
> Please also note that ```xxx => {}``` is not a common pattern in our code. We usually use ```xxx '{}'```.

OK - will change this.

> This does not qualify for an error logging as it is an expected path from an ESH perspective and not to be considered a bug on our end.

What would you suggest here - use INFO logging ?

Reply:
DEBUG. And setting a proper ThingStatus, which will anyhow be logged as INFO.